### ISSUE TITLE
Nodes in multilayer 3D plot are not embedded in the layer

### ISSUE THREAD

**Issue by alpatania** [xgi-org/xgi#436]
Right now nodes are plotted using mpl scatter, which keep them as a perfect circle even when in the tilted layer. we should maybe use a patch?

**Comment by thomasrobiglio** [comment]
I have a (super preliminary) version of a solution to this on my local xgi. This is done following [this](https://matplotlib.org/stable/gallery/mplot3d/pathpatch3d.html). I still have to work to make it consistent with all the rest... 

<img width="511" alt="Screenshot 2024-10-02 alle 16 46 46" src="https://github.com/user-attachments/assets/22bc4414-2f2d-4b68-96a5-bdaeb3f3a19a">

There is a minor problem: with this solution we are forced to use circles to draw the nodes, thus losing the possibility to pass a `node_shape` parameter. Do we want this?

If you give me the green light on this I will work on a proper PR :)

**Comment by maximelucas** [comment]
Thanks Thomas! I think it's not a major problem if we loose the ability of changing the node shape. 
Would we loose anything else?

**Comment by thomasrobiglio** [comment]
Yes I thought the same about the nodes' shape. What can be more relevant is the following:

I am struggling to have a node collection returned when calling the function (which then allows to have the colorbars and the cool stuff we have for the standard drawing function). 

This is because I need to create a separate `Circle` patch that is transformed into a 3D patch for each node (in each level) separately, and I have not managed to group these objects in a collection so far. I will keep looking for a solution, if you have ideas they are most welcome, I am a bit stuck--I have opened a draft pr #612
